mvlearn/cluster/mv_coreg_spectral.py

```diff
@@ -0,0 +1,290 @@
+        -------
+        la_eigs : array-like, shape(n_samples, n_clusters)
+            The top n_cluster eigenvectors of the normalized graph
+            laplacian.
+        '''
+
+        # Compute the normalized laplacian
+        d_mat = np.diag(np.sum(X, axis=1))
+        d_alt = np.sqrt(np.linalg.inv(d_mat))
+        laplacian = d_alt @ X @ d_alt
+
+        # Make the resulting matrix symmetric
+        laplacian = (laplacian + np.transpose(laplacian)) / 2.0
+
+        # Obtain the top n_cluster eigenvectors of the laplacian
+        u_mat, d_mat, _ = sp.sparse.linalg.svds(laplacian, k=self.n_clusters)
+        obj_val = np.sum(d_mat)
+
+        return u_mat, laplacian, obj_val
+
+    def fit(self, Xs):
+
+        r'''
+        Performs clustering on the multiple views of data.
+
+        Parameters
+        ----------
+
+        Xs : list of array-likes or numpy.ndarray
+            - Xs length: n_views
+            - Xs[i] shape: (n_samples, n_features_i)
+
+            This list must be of size n_views, corresponding to the number
+            of views of data. Each view can have a different number of
+            features, but they must have the same number of samples.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        '''
+
+        check_u_mats = list()
+
+        # Perform checks on inputted parameters and data
+        Xs = self._param_checks(Xs)
+        if self.v_lambda <= 0:
+            msg = 'v_lambda must be a positive value'
+            raise ValueError(msg)
+
+        # Compute the similarity matrices
+        sims = [self._affinity_mat(X) for X in Xs]
+
+        # Initialize matrices of eigenvectors
+        U_mats = []
+        L_mats = []
+        obj_vals = np.zeros((self._n_views, self.max_iter))
+        for ind in range(len(sims)):
+            u_mat, l_mat, o_val = self._init_umat(sims[ind])
+            U_mats.append(u_mat)
+            L_mats.append(l_mat)
+            obj_vals[ind, 0] = o_val
+
+        check_u_mats.append(U_mats[0])
+
+        # Iteratively solve for all U's
+        n_items = Xs[0].shape[0]
+        for it in range(1, self.max_iter):
+            # Performing alternating maximization by cycling through all
+            # pairs of views and updating all except view 1
+            for v1 in range(1, self._n_views):
+                # Computing the regularization term for view v1
+                l_comp = np.zeros((n_items, n_items))
+                for v2 in range(self._n_views):
+                    if v1 != v2:
+                        l_comp = l_comp + U_mats[v2] @ U_mats[v2].T
+                l_comp = (l_comp + l_comp.T) / 2
+                # Adding the symmetrized graph laplacian for view v1
+                l_mat = L_mats[v1] + self.v_lambda * l_comp
+                U_mats[v1], d_mat, _ = sp.sparse.linalg.svds(l_mat,
+                                                             k=self.n_clusters)
+                obj_vals[0, it] = np.sum(d_mat)
```

Comment:
is this correct? why is this being updated in the inner loop if the indices [0,it] only depend on the outer loop and the d_mat changes every time? Should it be [index_of_view, it] instead?

Reply:
You're right. It should be [v1, it], and that's what it previously was based on my validation notebooks. I think I had deleted it at some point and then copied and pasted the line from line 249 to add it back but forgot to change the indices.